### PR TITLE
TST: series/indexing tests parameterization + moving test methods

### LINKED ISSUE #20014: CLN: followons to refactor of series/test_indexing.py
xref #18164 

- [ ] remove the classes, IOW make all of the tests at function level. This will require removing the ``setup_method`` and making fixtures for this.
- [ ] some tests need to be moved around a bit, IOW we want to move tests out of test_indexing.py (the new one), so appropriate places (e.g. .drop should be in alter_axes). 
- [ ] want to parameterize things (e.g. anything with loops)

any/all of these can be separate PRs.

### PR DESCRIPTION
**Pull request by almaleksia**
Parts 2 and 3 of #20014. 
- tests parameterization
- moving some tests from test_indexing.py to other files 
- test methods reordering in test_indexing.py

- [x] closes #20014
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/series/indexing/test_boolean.py [hunk 1]
```diff
@@ -403,37 +399,32 @@ def f():
...
+    [False]
+])
+@pytest.mark.parametrize('item', [
+    2.0, np.nan, np.finfo(np.float).max, np.finfo(np.float).min
+])
+def test_where_broadcast(size, mask, item):
+    selection = np.resize(mask, size)
+    # Test numpy arrays, lists and tuples as the input to be
+    # broadcast
+    for arr in [np.array([item]), [item], (item,)]:
```
Comment (jreback) [inline pandas/tests/series/indexing/test_boolean.py]: could add this to the parameterization, we typically would call this ``box: [lambda x: np.array([x]), lambda x: [x], lambda x: (x,)]``

File: pandas/tests/series/indexing/test_boolean.py [hunk 2]
```diff
@@ -587,27 +578,27 @@ def test_mask():
...
+    [True, False],
+    [False]
+])
+@pytest.mark.parametrize('item', [
+    2.0, np.nan, np.finfo(np.float).max, np.finfo(np.float).min
+])
+def test_mask_broadcast(size, mask, item):
+        selection = np.resize(mask, size)
+
+        for arr in [np.array([item]), [item], (item,)]:
```
Comment (jreback) [inline pandas/tests/series/indexing/test_boolean.py]: same as above, though would not be averse to testing both mask and where in the same function to make this simpler

**Review by jreback — CHANGES_REQUESTED** [review]
very nice! just a couple of minor comments. ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
very nice @almaleksia !

**Comment by jreback** [comment]
if you wanted to move the JOIN_TYPES to the top-level ``pandas/conftest.py`` (and make it a fixture),  and prob rename to join_types, would be *really* helpful. note that this will percolate to lots of other modules, mainly in pandas/tests/reshape where they are actually used.

**Comment by jreback** [comment]
I *think* we have an issue about this (or at least about generally moving things like this to fixtures). Also would be great to split up pandas/tests/frame/test_indexing.py along the same lines as you did here :-D

**Comment by almaleksia** [comment]
@jreback thanks! I can create separate PRs for joins and frame/test_indexing refactoring

**Pull request merged**